src/component/GoodsList.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+type State = {
+  isReversed: boolean;
+  sortBy: string;
+};
+
+type Props = {
+  goods: string[];
+};
+
+export class GoodsList extends React.Component<Props, State> {
+  state = {
+    isReversed: false,
+    sortBy: '',
+  };
+
+  reverse = () => {
+    this.setState(currentState => ({
+      isReversed: !currentState.isReversed,
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortBy: 'alphabet' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
```

Comment:
you can make 1 function for sorting using parameters that you can pass into function